### PR TITLE
Tx Types and ERC20 token transaction

### LINKED ISSUE #5732: Example of EIP-1559 Transfer for ERC20 token
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to say about Web3.js' documentation?

This particular example of an EIP-1559 transaction to transfer ER20 token from one address to another, which should be very common (e.g. for sending USDC), is not covered in the docs and I googled the entire web with no results. Can you please help?

### Examples/References

_No response_

### PR DESCRIPTION
**Pull request by SantiagoDevRel**
Closes #5732

### CODE REVIEW

File: docs/docs/guides/wallet/tx-types.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
+---
+sidebar_position: 3
+sidebar_label: 'Transaction types'
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
sidebar_label: 'Transaction Types'
```

File: docs/docs/guides/wallet/tx-types.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+---
+
+# Transactions
+
+In this tutorial, we will explore how to send different types of [transactions](https://ethereum.org/en/developers/docs/transactions/) using web3.js, focusing on Ethereum's evolving transaction formats. We'll start with [legacy transactions (Transaction Type 0)](#transaction-type-0-legacy). Next, we'll delve into Transaction [Type 1 (EIP-2930)](#transaction-type-1-eip-2930), which introduces access lists to optimize gas usage. Finally, we'll cover [Transaction Type 2 (EIP-1559)](#transaction-type-2-eip-1559), the current default, which allows users to specify maximum fees and priority tips for more efficient and cost-effective transactions. Each section will include practical code examples to demonstrate sending raw transactions and interacting with ERC20 tokens on the Sepolia test network
+
+:::note
+Web3.js uses transactions type 2 by default
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
Web3.js uses transaction type 2 by default
```

File: docs/docs/guides/wallet/tx-types.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+:::
+
+## Transaction Type 0 (Legacy)
+
+### Raw Transaction 
+
+A Legacy Transaction refers to a transaction that was created using an older version of Ethereum's transaction format, also known as "transaction type 0". This transaction format was used before the EIP-1559 upgrade, which was implemented in August 2021.
+
+```ts
+import { Web3 } from"web3";
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
import { Web3 } from "web3";
```

File: docs/docs/guides/wallet/tx-types.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+const web3 = new Web3("https://rpc2.sepolia.org");
+
+async function txLegacy() {
+  const wallet = web3.eth.wallet.add("YOUR_PRIVATE_KEY"); //make sure you have funds
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]: All comments should have a space after the slashes.

```suggestion
  const wallet = web3.eth.wallet.add("YOUR_PRIVATE_KEY"); // make sure you have funds
```

File: docs/docs/guides/wallet/tx-types.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+```
+
+## Transaction Type 1 (EIP-2930)
+
+This EIP was introduced in April 2021, it introduces a feature called 'Access List.' This improvement allows saving gas on cross-contract calls by declaring in advance which contract and storage slots will be accessed.
+
+### Raw Transaction
+
+```ts
+import { Web3 } from"web3";
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
import { Web3 } from "web3";
```

File: docs/docs/guides/wallet/tx-types.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+```
+
+## Transaction Type 2 (EIP-1559)
+
+When a user creates an EIP-1559 transaction, they specify the maximum fee they are willing to pay `maxFeePerGas` as well as a tip `maxPriorityFeePerGas` to incentivize the miner. The actual fee paid by the user is then determined by the network based on the current demand for block space and the priority of the transaction.
+
+### Raw Transaction
+
+```ts
+import { Web3 } from"web3";
```
Comment (danforbes) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
import { Web3 } from "web3";
```

File: docs/docs/guides/wallet/tx-types.md (hunk 3 above)
Comment (luu-alex) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
import { Web3 } from "web3";
```

File: docs/docs/guides/wallet/tx-types.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+const web3 = new Web3("https://rpc2.sepolia.org");
+
+async function txLegacy() {
+  const wallet = web3.eth.wallet.add("YOUR_PRIVATE_KEY"); //make sure you have funds
+
+  const sender = wallet[0].address;
+  const recipient = "0x807BFe4940016B5a7FdA19482042917B02e68359";
+  const value = 1; //wei
```
Comment (luu-alex) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
  const value = 1; // wei
```
comments should always have a space after the //, please update this for all other comments

File: docs/docs/guides/wallet/tx-types.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+  
+  console.log("Tx hash", txReceipt.transactionHash);
+}
+
+txEIP2930()
+```
+
+## Transaction Type 2 (EIP-1559)
+
+When a user creates an EIP-1559 transaction, they specify the maximum fee they are willing to pay `maxFeePerGas` as well as a tip `maxPriorityFeePerGas` to incentivize the miner. The actual fee paid by the user is then determined by the network based on the current demand for block space and the priority of the transaction.
```
Comment (luu-alex) [inline docs/docs/guides/wallet/tx-types.md:183]: Very helpful! nice.

**Comment by SantiagoDevRel** [comment]
Thank you @danforbes for the heads up w the typos!💪

**Comment by luu-alex** [comment]
if you can also loop @danforbes  into this that would be great. When he has free time would like to also have his opinion on this!

**Comment by luu-alex** [comment]
nevermind, hes already commented :)

File: docs/docs/guides/wallet/tx-types.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+## Transaction Type 0 (Legacy)
+
+### Raw Transaction 
+
+A Legacy Transaction refers to a transaction that was created using an older version of Ethereum's transaction format, also known as "transaction type 0". This transaction format was used before the EIP-1559 upgrade, which was implemented in August 2021.
+
+```ts
+import { Web3 } from "web3";
+
+const web3 = new Web3("https://rpc2.sepolia.org");
```
Comment (luu-alex) [inline docs/docs/guides/wallet/tx-types.md]:
```suggestion
const web3 = new Web3("https://rpc2.sepolia.org"); // RPC node url
```
think its nice to have this comment

**Review by luu-alex — APPROVED** [review]

**Pull request merged**